File: Samples/word-blazor-add-in/word-blazor-add-in/Pages/HelloWorld.razor.cs
/* Copyright(c) Maarten van Stam. All rights reserved. Licensed under the MIT License. */

using Microsoft.AspNetCore.Components;
using Microsoft.JSInterop;

namespace BlazorAddIn.Pages
{
    /// <summary>
    /// Starter class to demo how to insert a paragraph
    /// </summary>
    public partial class HelloWorld
    {
        [Inject]
        public IJSRuntime JSRuntime { get; set; } = default!;
        public IJSObjectReference JSModule { get; set; } = default!;

        protected override async Task OnAfterRenderAsync(bool firstRender)
        {
            if (firstRender)
            {
                JSModule = await JSRuntime.InvokeAsync<IJSObjectReference>("import", "./Pages/HelloWorld.razor.js");
            }
        }

        /// <summary>
        /// Basic function to invoke inserting a paragraph
        /// </summary>
        private async Task InsertParagraph() =>
            await JSModule.InvokeVoidAsync("insertParagraph");
    }
}

File: Samples/word-blazor-add-in/word-blazor-add-in/Pages/HelloWorld.razor.js
/* Copyright(c) Maarten van Stam. All rights reserved. Licensed under the MIT License. */
/**
 * Basic function to show how to insert a Paragraph at the start of the Word Document
 */
export function insertParagraph() {

    console.log("Hello JavaScript in Blazor!?!?!?");

    return Word.run((context) => {

        // insert a paragraph at the start of the document.
        const paragraph = context.document.body.insertParagraph("Hello World from Page2.razor.js", Word.InsertLocation.start);

        // sync the context to run the previous API call, and return.
        return context.sync();
    });
}
